Document MacOS quirk wrt DYLD_LIBRARY_PATH and SIP

patrickbkr opened the pull request:
MacOS has some annoying behavior when dynamically loading C libs. Document those.
See https://github.com/Raku/App-Rakubrew/issues/9 for the very long story.

zaucker commented:
I wonder who the target audience for this description is. While it helps me (as a Rakudo user) understand that there is an issue with SIP, it doesn't really tell me what I have to do (if anything) when installing and using it.

vrurg commented:
> it doesn't really tell me what I have to do (if anything) when installing and using it.

Because there is no single solution for all. For example, installing `rakudo` into standard MacPorts or Homebrew paths would not be affected by the problem due to how SIP is implemented.

vrurg commented on doc/Language/nativecall.pod6:
```diff
@@ -1246,6 +1246,20 @@ AF_INET6 SOCK_RAW
+I<shebang> lines, is one of those programs. This means that whenever C<env> is
+involved in calling a program the C<DYLD_LIBRARY_PATH> variable will be
+cleared. To work around this effect one has to either make sure no protected
+process is involved (this can be difficult) or disable SIP.
```
Sorry for being late. I missed your request for review.

I think we need an addition here stating that custom installations of Rakduo into non-standard paths, like under one's home directory, are the most affected ones. But more details in the link below.

Otherwise looks good. Thanks!

zaucker commented:
> > it doesn't really tell me what I have to do (if anything) when installing and using it.
> 
> Because there is no single solution for all. For example, installing `rakudo` into standard MacPorts or Homebrew paths would not be affected by the problem due to how SIP is implemented.

I currently use the binary releases from https://rakudo.org/downloads unpacking the tar-file into `$HOME/opt/` and symlink that to `$HOME/opt/rakudo`. My PATH then includes `$HOME/opt/rakudo/bin:$HOME/opt/rakudo/share/perl6/site/bin`.
This seems to work fine so far, but of course I might miss some usages that might fail.

This is **not** meant to criticise the suggested documentation, just trying to let you know what some ordinary user might (not) get from it.

patrickbkr commented:
@vrurg @zaucker I added a paragraph listing situations that tend to trigger the problem and added another link to a good overview of the problem (that google only showed me *after* I found out about the offender and added "SIP" to the search terms. Grrr.)

vrurg commented:
Looks good. Thank you, Patrick!
